examples/step-89/step-89.cc

```diff
@@ -0,0 +1,1884 @@
+    template <typename VectorType>
+    void local_apply_boundary_face_mortaring(
+      const MatrixFree<dim, Number>               &matrix_free,
+      VectorType                                  &dst,
+      const VectorType                            &src,
+      const std::pair<unsigned int, unsigned int> &face_range) const
+    {
+      // Standard face evaluators for BCs.
+      FEFaceEvaluation<dim, -1, 0, 1, Number> pressure_m(
+        matrix_free, true, 0, 0, 0);
+      FEFaceEvaluation<dim, -1, 0, dim, Number> velocity_m(
+        matrix_free, true, 0, 0, 1);
+
+      // For Nitsche-type mortaring we are evaluating the integrals over
+      // intersections. This is why, quadrature points are arbitrarely
+      // distributed on every face. Thus, we can not make use of face batches
+      // and FEFaceEvaluation but have to consider each face individually and
+      // make use of @c FEPointEvaluation to evaluate the integrals in the
+      // arbitrarely distributed quadrature points.
+      // TODO: The setup of FEPointEvaluation is more
+      //  expensive than that of FEEvaluation (since MatrixFree stores
+      //  all the precomputed data). Should we use ThreadLocalStorage?
+      FEPointEvaluation<1, dim, dim, Number> pressure_m_mortar(
+        *nm_mapping_info, matrix_free.get_dof_handler().get_fe(), 0);
+      FEPointEvaluation<dim, dim, dim, Number> velocity_m_mortar(
+        *nm_mapping_info, matrix_free.get_dof_handler().get_fe(), 1);
+      AlignedVector<Number> buffer(
+        matrix_free.get_dof_handler().get_fe().dofs_per_cell);
+
+      BCEvalP pressure_bc(pressure_m);
+      BCEvalU velocity_bc(velocity_m);
+
+      MaterialEvaluation material(matrix_free, *material_data);
+
+      auto pressure_r_mortar = pressure_r_eval->get_data_accessor();
+      auto velocity_r_mortar = velocity_r_eval->get_data_accessor();
+      auto c_r               = c_r_eval->get_data_accessor();
+      auto rho_r             = rho_r_eval->get_data_accessor();
+
+      for (unsigned int face = face_range.first; face < face_range.second;
+           ++face)
+        {
+          if (HelperFunctions::is_non_matching_face(
+                remote_face_ids, matrix_free.get_boundary_id(face)))
+            {
+              // For mortaring, we have to cosider every face from the face
+              // batches seperately and have to use the FEPointEvaluation
+              // objects to be able to evaluate the integrals with the
+              // arbitrarily distributed quadrature points.
+              for (unsigned int v = 0;
+                   v < matrix_free.n_active_entries_per_face_batch(face);
+                   ++v)
+                {
+                  const auto [cell, f] =
+                    matrix_free.get_face_iterator(face, v, true);
+
+                  // TODO: we will be able to simplify this in a follow up PR
+                  // once there is FEFacePointEvaluation.
```

Comment:
Maybe convert these inline TODOs to a TODO list?

Reply:
:+1: I will do this before merging. While working on the PR, I like the inline TODOs. It's easier for me and I assume it is easier to review and comment on them.